hpi-swt2/lecture-portal
# Feature/#125 upload links

# Resolves #125 , #159

This PR  introduces the feature of uploading links as if they were files.

Acceptance Criteria:
From #125 :
- [x] The DB model should support a file's content to be either an actual file stored in the DB, OR to be a hyperlink. Not both!
- Additionally, I implemented a rudimentary UI including functional tests.  
From #159 :
- [x] The lectures and the students dashboard both contain a section called "Lecture materials"
- [x] This section holds a scrollable list of all files associated to this particular lecture
- [x] Clicking on a files name will "open" it (download if file / open link if link)


Testing steps:
- go to http://localhost:3000/uploaded_files
- click "Upload file"
- enter a valid lecture ID and a Hyperlink into the text field
- click the button
- you should see the hyperlink on the next page

Steps needed for migration:
- DB migrations

---

Definition Of Done:
